### PR TITLE
add Indonesian newspapers (id_newspapers_2018)

### PR DESCRIPTION
**Pull request by cahya-wirawan**
The dataset contains around 500K articles (136M of words) from 7 Indonesian newspapers. The size of uncompressed 500K json files (newspapers-json.tgz) is around 2.2GB.

### CODE REVIEW

**Comment by yjernite** [comment]
Looks like there's a `Path` issue on windows. Could you try switching to
`glob.glob(os.path.join(article_dir, "*.json"))`

**Comment by cahya-wirawan** [comment]
> Looks like there's a `Path` issue on windows. Could you try switching to
> `glob.glob(os.path.join(article_dir, "*.json"))`

Thanks, I replaced it with glob. Let's see if it solves the issue. Anyway, the main directory has a space, could it make the issue on windows? the test on linux don't have this problem.

**Comment by cahya-wirawan** [comment]
It seems glob doesn't help also

**Comment by cahya-wirawan** [comment]
It seems that the circleci make a test for whole datasets repository, that means if only one of the dataset in the official repository has a download issue, this will also affect the test of a new dataset like mine, isn't it?
I changed the url to my newspaper dataset which contains only few simple json files and simple directory structure. But it still failed. And it failed not only on windows test. This is one of the error message:
```
-- Docs: https://docs.pytest.org/en/stable/warnings.html
=========================== short test summary info ============================
FAILED tests/test_dataset_common.py::LocalDatasetTest::test_load_dataset_ajgt_twitter_ar
FAILED tests/test_dataset_common.py::LocalDatasetTest::test_load_dataset_chr_en
FAILED tests/test_dataset_common.py::RemoteDatasetTest::test_load_dataset_ajgt_twitter_ar
FAILED tests/test_dataset_common.py::RemoteDatasetTest::test_load_dataset_chr_en
===== 4 failed, 2667 passed, 2052 skipped, 4 warnings in 432.05s (0:07:12) =====

Exited with code exit status 1
CircleCI received exit code 1
```
The test failed on twitter dataset even my dataset has nothing to do with twitter?

File: datasets/id_newspapers_2018/README.md [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+language_creators:
+- found
+languages:
+- id
+licenses:
+- other-my-license
+multilinguality:
+- monolingual
+size_categories:
+- n>100M
```
Comment (lhoestq) [inline datasets/id_newspapers_2018/README.md:13]:
```suggestion
- n>1M
```
we haven't added a tag for higher values than 1M yet ^^

File: datasets/id_newspapers_2018/id_newspapers_2018.py [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                    "article_dir": os.path.join(data_dir, "newspapers"),
+                    "split": "train",
+                },
+            )
+        ]
+
+    def _generate_examples(self, article_dir, split):
+        logging.info("⏳ Generating %s examples from = %s", split, article_dir)
+        id = 0
+        for path in glob.glob(os.path.join(article_dir, "**/*.json"), recursive=True):
```
Comment (lhoestq) [inline datasets/id_newspapers_2018/id_newspapers_2018.py:109]:
```suggestion
        for path in sorted(glob.glob(os.path.join(article_dir, "**/*.json"), recursive=True)):
```
we have to make the order deterministic here

**Review by lhoestq — APPROVED** [review]
Awesome thank you :) LGTM

**Comment by lhoestq** [comment]
merging since the CI is fixed on master

**Comment by cahya-wirawan** [comment]
Hi, thanks for merging the dataset. I create a new PR (#1499) since I need to update the link to the dataset.

**Pull request merged**